### ISSUE TITLE
✨ Work email - new pattern and let support team save without validation

### ISSUE THREAD

**Issue by gobyrne** [GCTC-NTGC/gc-digital-talent#12027]
## ✨ Feature

We've gotten two new email patterns (`@ccc.ca` and `@canadapost.postescanada.ca` and '@gg.ca') for a government employee that we need to whitelist. So let's add it.

Also, we want to make a bunch of stuff blocked by Government email verification going forward. For example, you can't apply to an internal process, and you can't unlock employee tools.

Before we do that, we want support team to be able to immediately help anyone who flags that their email doesn't validate. This will allow that user to complete applying to a job (for example).

A simple way to achieve this is to allow platform admins to save a work email for an employee that does not pass our current validation. Then the user will have it saved and can simply to the email verification. This will require a new mutation that requires platform_admin permission but bypasses normal validation.

## Proposed solution

Update `UpdateUserAsAdminInput` so that it uses its own validator, UpdateUserAsAdminInputValidator. This should be identical to UpdateUserInputValidator _except_ that it doesn't include the endswith whitelist for work email. Ensure the new class has comments explaining the difference.

## ✅ Acceptance Criteria

- [x] Double check if `@canadapost.postescanada.ca` uses a dot or a hyphen
- [x] Add `@ccc.ca` and `@canadapost.postescanada.ca` to our whitelist for government emails.
- [x] Remove email pattern validation from Platform Admin UI
- [ ] When a Platform Admin updates a user, it uses a different validator which doesn't require a whitelisted work email

**Comment by tristan-orourke** [comment]
Is it `@canadapost.postescanada.ca` or `@canadapost-postescanada.ca`? Lets double check

**Comment by marc-donofrio** [comment]
We got another request for: @gg.ca ( Governor General's Office)

**Comment by DVPETROVA** [comment]
Good morning. Another email to add to this list please: @servicecanada.gc.ca

**Comment by tristan-orourke** [comment]
While we know that this isn't an ideal solution for users who run into not-allowed emails in the future, this doesn't make anything worse for users, so we will go ahead with it now as is.

**Comment by DVPETROVA** [comment]
Another one to add please - @pmo-cpm.gc.ca

**Comment by petertgiles** [comment]
> @pmo-cpm.gc.ca

This is a subdomain of gc.ca and so it shouldn't need its own rule, like tbs-sct.gc.ca.  Also, this issue was closed yesterday and so no more changes can be made on this one.

**Comment by marc-donofrio** [comment]
To be added to:
@humanrights.ca 

Canadian Museum for Human rights

**Comment by petertgiles** [comment]
@marc-donofrio Could you please open new issue for the request?   This issue is closed and can't be changed.
